test/wallaby/http_client_test.exs

```diff
@@ -73,7 +73,7 @@ defmodule Wallaby.HTTPClientTest do
 
     test "includes the original HTTPoison error when there is one", %{bypass: bypass} do
       expected_message =
-        "Wallaby had an internal issue with HTTPoison:\n%HTTPoison.Error{id: nil, reason: :econnrefused}"
+        "Wallaby had an internal issue with HTTPoison:\n%HTTPoison.Error{reason: :econnrefused, id: nil}"
```

Comment:
if this is flaky, i would do multiple `=~` asserts instead

Reply:
I'm not confident that it is flaky just yet. I want to understand why this change was necessary.

Reply:
otp changed how maps work for keys < 32 so now key order is not deterministic like it used to be

Reply:
Agreed. But if maps < 32 keys are treated like keyword lists in memory, they are linked and therefore have a deterministic order yes? So my inference would be that `{:id, nil}` was added before `{:reason, :econnrefused}` in the internals of HTTPoison, thereby putting `:reason` at the head of the linked list.

Reply:
I'm trying to say that in OTP26 that is no longer the case. the order is nondeterministic

https://www.erlang.org/blog/otp-26-highlights/#changed-ordering-of-atom-keys

You can pass `custom_options: [sort_maps: true]` to as an Inspect.Opt tho, which will sort it and it should be deterministic.

Reply:
Ah. I see! Thanks for explaining 😃 I'll make some more adjustments and see if I can get things backwards compatible as well. Thanks for the feedback!

Reply:
Of course 😁